rascal-lsp/src/test/java/org/rascalmpl/vscode/lsp/rascal/model/PathConfigDiagnosticsTest.java

```diff
@@ -0,0 +1,171 @@
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ISourceLocation resourceLocation(String res, int line, int col) {
+        ISourceLocation loc = resourceLocation(res);
+        int offset = columnMaps.get(loc).calculateInverseOffsetLength(line, col, line, col).getLeft();
+        return VF.sourceLocation(loc, offset, 1, line, line, col, col);
+    }
+
+    private String getContents(ISourceLocation file) {
+        try (Reader src = URIResolverRegistry.getInstance().getCharacterReader(file)) {
+            return Prelude.consumeInputStream(src);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private IConstructor msg(String file, String msg, int line, int col) {
+        return Messages.info(msg, resourceLocation(file, line, col));
+    }
+
+    private Diagnostic diag(int line, int col, String msg) {
+        var diagnostic = new Diagnostic(new Range(new Position(line, col), new Position(line, col)), msg);
+        diagnostic.setSeverity(DiagnosticSeverity.Information);
+        return diagnostic;
+    }
+
+    private PublishDiagnosticsParams params(String url, Diagnostic... diags) {
+        return new PublishDiagnosticsParams(url, Arrays.asList(diags));
+    }
+
+    @Test
+    public void testPublishFresh() {
+        sut.publishDiagnostics(projectA, VF.list(msg(POM_A, MSG1, 4, 7)));
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(3, 7, MSG1)));
+    }
+
+    @Test
+    public void testPublishMultiProject() {
+        sut.publishDiagnostics(projectA, VF.list(msg(POM_A, MSG1, 4, 7)));
+        // Expect the diagnostics for project A
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(3, 7, MSG1)));
+
+        sut.publishDiagnostics(projectB, VF.list(msg(POM_A, MSG2, 5, 7)));
+        // Expect the diagnostics for both projects
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(3, 7, MSG1), diag(4, 7, MSG2)));
+    }
+
+    @Test
+    public void testDeleteOursButRetainTheirs() {
+        sut.publishDiagnostics(projectA, VF.list(msg(POM_A, MSG1, 4, 7)));
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(3, 7, MSG1)));
+
+        sut.publishDiagnostics(projectB, VF.list(msg(POM_A, MSG2, 5, 7)));
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(3, 7, MSG1), diag(4, 7, MSG2)));
+
+        reset(mockedClient);
+        sut.publishDiagnostics(projectA, VF.list());
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(4, 7, MSG2)));
+    }
+
+    @Test
+    public void testClearDiagnostics() {
+        sut.publishDiagnostics(projectA, VF.list(msg(POM_A, MSG1, 4, 7)));
+        sut.publishDiagnostics(projectB, VF.list(msg(POM_A, MSG2, 5, 7)));
+
+        reset(mockedClient);
+        sut.clearDiagnostics(projectA);
+        verify(mockedClient).publishDiagnostics(params(pomAUrl, diag(4, 7, MSG2)));
+    }
```

Comment:
Nice that we have detailed tests. We don't have them for any other interaction in the system, and many places it's hard to mock, as a lot of context needs to be setup.

So nice, but a set of UI test would have been enough I think, at least to match the rest of the tests.